-1

Tested at: ebf87d8f9b34eac16a498a9a8f721dd3e03b7d3d

CMSSW: CMSSW_11_1_X_2020-07-16-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-916add/8037/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/DBOutputService/src/OnlineDBOutputService.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/DBOutputService/src/PoolDBOutputService.cc 
>> Building shared library tmp/slc7_amd64_gcc820/src/CondCore/DBOutputService/src/CondCoreDBOutputService/libCondCoreDBOutputService.so
/cvmfs/cms-ib.cern.ch/nweek-02637/slc7_amd64_gcc820/external/gcc/8.2.0-bcolbf/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.4.0/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc820/src/CondCore/DBOutputService/src/CondCoreDBOutputService/OnlineDBOutputService.cc.o: in function `cond::service::OnlineDBOutputService::OnlineDBOutputService(edm::ParameterSet const&, edm::ActivityRegistry&)':
OnlineDBOutputService.cc:(.text+0x1d7a): undefined reference to `cond::persistency::Session::getLastRun()'
collect2: error: ld returned 1 exit status
gmake: *** [tmp/slc7_amd64_gcc820/src/CondCore/DBOutputService/src/CondCoreDBOutputService/libCondCoreDBOutputService.so] Error 1
Leaving library rule at CondCore/DBOutputService
------- copying files from src/CondCore/DBOutputService/scripts -------
>> copied cmscond_logdb_dump
Entering library rule at src/CondCore/DBOutputService/test
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/CondDB/src/Time.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/CondDB/src/Types.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/CondDB/src/base64.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/CondDB/src/blowfish.cc 
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/CondDB/src/RunInfoSchema.cc:2:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/CondDB/src/RunInfoSchema.h:46:40: error: non-virtual member function marked 'override' hides virtual member function
        cond::Time_t getLastInserted() override;
                                       ^
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-07-16-2300/src/CondCore/CondDB/src/IDbSchema.h:222:28: note: hidden overloaded virtual function 'cond::persistency::IRunInfoTable::getLastInserted' declared here: different number of parameters (2 vs 0)
      virtual cond::Time_t getLastInserted(boost::posix_time::ptime& start, boost::posix_time::ptime& end) = 0;
                           ^
</pre>

--- a/CondCore/CondDB/src/RunInfoSchema.h
+++ b/CondCore/CondDB/src/RunInfoSchema.h
@@ -43,7 +43,7 @@ namespace cond {
         bool exists() override;
         void create() override;
         bool select(cond::Time_t runNumber, boost::posix_time::ptime& start, boost::posix_time::ptime& end) override;
-        cond::Time_t getLastInserted() override;
+        cond::Time_t getLastInserted(boost::posix_time::ptime& start, boost::posix_time::ptime& end) override;
         bool getInclusiveRunRange(
             cond::Time_t lower,
             cond::Time_t upper,

